realm/src/main/java/io/realm/dynamic/DynamicRealmObject.java

```diff
@@ -0,0 +1,219 @@
+        List<String> fields = new ArrayList<String>();
+        long columns = row.getColumnCount();
+        for (int i = 0; i < columns; i++) {
+            fields.add(row.getColumnName(i));
+        }
+        return fields;
+    }
+
+    public boolean hasField(String fieldName) {
+        if (fieldName == null || fieldName.isEmpty()) {
+            return false;
+        }
+        return row.getColumnIndex(fieldName) != TableOrView.NO_MATCH;
+    }
+
+    public String[] getFieldNames() {
+        String[] keys = new String[(int) row.getColumnCount()];
+        for (int i = 0; i < keys.length; i++) {
+            keys[i] = row.getColumnName(i);
+        }
+        return keys;
+    }
+
+    private void checkFieldExists(long columnIndex, String fieldName) {
+        if (columnIndex == TableOrView.NO_MATCH) {
+            throw new IllegalArgumentException("'" + fieldName + "' doesn't exist on " + row.getTable().getName());
+        }
+    }
+
+    private void checkColumnType(ColumnType expectedColumnType, long columnIndex, String fieldName) {
+        ColumnType columnType = row.getColumnType(columnIndex);
+        if (columnType != expectedColumnType) {
+            throw new IllegalArgumentException(fieldName + " is not the expected type. It is a " + columnType);
+        }
+    }
+
+    private void checkLinkedField(String fieldName) {
+        if (fieldName.contains("\\.")) {
+            throw new IllegalArgumentException("Fetching data across links not supported yet.");
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        String realmName = realm.getPath();
+        String tableName = row.getTable().getName();
+        long rowIndex = row.getIndex();
+
+        int result = 17;
+        result = 31 * result + ((realmName != null) ? realmName.hashCode() : 0);
+        result = 31 * result + ((tableName != null) ? tableName.hashCode() : 0);
+        result = 31 * result + (int) (rowIndex ^ (rowIndex >>> 32));
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        DynamicRealmObject other = (DynamicRealmObject) o;
+
+        String path = realm.getPath();
+        String otherPath = other.realm.getPath();
+        if (path != null ? !path.equals(otherPath) : otherPath != null) {
+            return false;
+        }
+
+        String tableName = row.getTable().getName();
+        String otherTableName = other.row.getTable().getName();
+        if (tableName != null ? !tableName.equals(otherTableName) : otherTableName != null) {
+            return false;
+        }
+
+        if (row.getIndex() != other.row.getIndex()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        return super.toString(); // TODO How to iterate across all fields?
```

Comment:
Smart ass :) My problem was accessing the data since AFAIK there is no generic row.getObject(index) function, so you would have to check the column type for each index in order to use the proper getX function. It's doable sure, but seemed like a lot of work if we had a smarter function. Maybe using the json() function in core?